src/librustc_lint/builtin.rs

```diff
@@ -272,9 +278,9 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UnsafeCode {
         }
     }
 
-    fn check_trait_item(&mut self, cx: &LateContext, item: &hir::TraitItem) {
-        if let hir::TraitItemKind::Method(ref sig, hir::TraitMethod::Required(_)) = item.node {
-            if sig.header.unsafety == hir::Unsafety::Unsafe {
+    fn check_trait_item(&mut self, cx: &EarlyContext, item: &ast::TraitItem) {
+        if let ast::TraitItemKind::Method(ref sig, _) = item.node {
```

Comment:
```suggestion
        if let ast::TraitItemKind::Method(ref sig, None) = item.node {
```

We should only lint the declaration if there's no default body for the method.

Reply:
OK, I changed.